osquery/tables/system/linux/disk_encryption.cpp

```diff
@@ -0,0 +1,100 @@
+#include <unistd.h>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/sql.h>
+
+extern "C" {
+#include <libcryptsetup.h>
+
+crypt_status_info crypt_status(struct crypt_device *cd, const char *name);
```

Comment:
I kept getting `undefined reference` linker errors. I think the names were getting mangled so put them all in extern C block.